src/routeStateSanityCheck.ts

```diff
@@ -0,0 +1,137 @@
+import { ConnectionMetadata, ItemType, RouteState, RouteStop } from './types'
+
+export const checkNumberOfRouteStateElementsEqualRouteElementNumber = ({
+  routeState,
+  routeElements
+}: {
+  routeState: RouteState
+  routeElements: RouteStop[]
+}): boolean => routeState.length === routeElements.length
+
+export const checkNumberOfConnectionsMatchesRouteElementsNumber = ({
+  routeElements,
+  connections
+}: {
+  routeElements: RouteStop[]
+  connections: ConnectionMetadata[]
+}): boolean => {
+  const routeElementsWithoutSections = routeElements.filter(
+    ({ itemType }) => itemType !== ItemType.Section
+  )
+  return connections.length === routeElementsWithoutSections.length - 1
+}
+
+export const checkRouteStateHasNoHoles = ({
+  routeState
+}: {
+  routeState: RouteState
+}): boolean =>
+  routeState
+    .sort((a, b) => (a.position > b.position ? 1 : -1))
+    .reduce(
+      (result: boolean, { position }, index) => result && index === position,
+      true
+    )
```

Comment:
I still don't understand why the order of the elements inside the route state is relevant. Can you please explain? Maybe with an example?

Reply:
It is not the order that is relevant, but the fact that there are no holes. In this case our whole logic on where to add new elements is broken. The state can be broken if the user has an older route which was created in the beginning when adding ferry connections.